### DISCUSSION TITLE
Image Magick - Cropping picture into circle

### DISCUSSION THREAD

**Discussion by webbooo** [ImageMagick/ImageMagick#5120] [Help] [answered]
I have 100+ png files all 500x500, I'm trying to crop a circle in the centre of the image 400x400 and save it into a separate png, keeping it at 500x500. example below

Does anyone know how to use image magick to achieve this and run through each file in the folder? - thanks

Source

![samplea](https://user-images.githubusercontent.com/103241119/166877964-7bfb9f7d-62a8-4c5e-a947-1438e61fca8a.png)

Desired outcome

![sampleb](https://user-images.githubusercontent.com/103241119/166878005-2e344801-d3bd-47df-9117-286f5117b254.png)

regards

**Comment by webbooo** [comment]
I have managed to use the below but unsure how to make it go through each file in the folder.  500x500 image with a 400px opaque circle in it was used for mask.png

`convert source.png mask.png -compose copy-opacity -composite result.png`

**Comment by snibgo** [comment, marked as answer]
> ... and run through each file in the folder? - thanks

That depends on the shell. For Windows BAT, see the example at https://github.com/ImageMagick/ImageMagick/discussions/5075

**Comment by fmw42** [comment]
What is your version of Imagemagick and your platform/OS?
